api/src/utils/validate-query.ts

```diff
@@ -42,12 +42,14 @@ export function validateQuery(query: Query): Query {
 	return query;
 }
 
-function validateFilter(filter: Query['filter']) {
+function validateFilter(filter: Query['filter'], isNested = false) {
 	if (!filter) throw new InvalidQueryException('Invalid filter object');
-
 	for (const [key, nested] of Object.entries(filter)) {
+		if (key === '_some' || key === '_none') {
+			if (isNested) throw new InvalidQueryException('Invalid filter object');
```

Comment:
I think this error can be more specific along the lines of invalid usage of `_some` or `_none`.